test/envtest/sgxdeviceplugin_controller_test.go

```diff
@@ -88,4 +93,41 @@ var _ = Describe("SgxDevicePlugin Controller", func() {
 			}, timeout, interval).ShouldNot(Succeed())
 		})
 	})
+
+	It("upgrades", func() {
+		name := "sgxdeviceplugin-test"
+		image0 := "intel/intel-sgx-plugin:0.22.0"
+		image := "intel/intel-sgx-plugin:" + controllers.ImageMinVersion.String()
+		client := k8sManager.GetClient()
+
+		dp0 := &devicepluginv1.SgxDevicePlugin{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: devicepluginv1.SgxDevicePluginSpec{
+				Image: image0,
+			},
+		}
+
+		Expect(client.Create(ctx, dp0) == nil)
+
+		time.Sleep(time.Second)
+
+		down()
+
+		up("sgx", true)
```

Comment:
Why this restart is needed? Can we have the operator running always, then create a plugin with an outdated image and check it again if it has the image field updated?